File: monopoly_engine/board.py
"""Board definitions for the Monopoly game engine.

This module defines the static board layout including all 40 spaces.
All definitions are frozen dataclasses for immutability.
"""

from dataclasses import dataclass
from typing import ClassVar

from .types import PROPERTY_GROUPS, PropertyColor, SpaceType


@dataclass(frozen=True)
class Space:
    """Base class for all board spaces."""

    position: int
    name: str
    space_type: SpaceType

    def to_dict(self) -> dict[str, object]:
        """Serialize to JSON-compatible dict."""
        return {
            "position": self.position,
            "name": self.name,
            "space_type": self.space_type.name,
        }


@dataclass(frozen=True)
class PropertySpace(Space):
    """A property that can be bought, built on, and rented."""

    color: PropertyColor
    cost: int
    rent: tuple[int, ...]  # (base, 1h, 2h, 3h, 4h, hotel)
    house_cost: int
    mortgage_value: int

    def to_dict(self) -> dict[str, object]:
        """Serialize to JSON-compatible dict."""
        base = super().to_dict()
        base.update(
            {
                "color": self.color.name,
                "cost": self.cost,
                "rent": list(self.rent),
                "house_cost": self.house_cost,
                "mortgage_value": self.mortgage_value,
            }
        )
        return base


@dataclass(frozen=True)
class RailroadSpace(Space):
    """A railroad property."""

    cost: int
    rent: tuple[int, int, int, int]  # (1 owned, 2 owned, 3 owned, 4 owned)
    mortgage_value: int

    def to_dict(self) -> dict[str, object]:
        """Serialize to JSON-compatible dict."""
        base = super().to_dict()
        base.update(
            {
                "cost": self.cost,
                "rent": list(self.rent),
                "mortgage_value": self.mortgage_value,
            }
        )
        return base


@dataclass(frozen=True)
class UtilitySpace(Space):
    """A utility property (Electric Company or Water Works)."""

    cost: int
    mortgage_value: int

    def to_dict(self) -> dict[str, object]:
        """Serialize to JSON-compatible dict."""
        base = super().to_dict()
        base.update(
            {
                "cost": self.cost,
                "mortgage_value": self.mortgage_value,
            }
        )
        return base


@dataclass(frozen=True)
class TaxSpace(Space):
    """A tax space (Income Tax or Luxury Tax)."""

    amount: int

    def to_dict(self) -> dict[str, object]:
        """Serialize to JSON-compatible dict."""
        base = super().to_dict()
        base["amount"] = self.amount
        return base


class Board:
    """The Monopoly board definition.

    This class provides access to all 40 board spaces and utility methods
    for querying the board layout.
    """

    # All 40 board spaces
    SPACES: ClassVar[tuple[Space, ...]] = (
        # Bottom row (right to left)
        Space(0, "GO", SpaceType.GO),
        PropertySpace(
            1,
            "Mediterranean Avenue",
            SpaceType.PROPERTY,
            PropertyColor.BROWN,
            60,
            (2, 10, 30, 90, 160, 250),
            50,
            30,
        ),
        Space(2, "Community Chest", SpaceType.COMMUNITY_CHEST),
        PropertySpace(
            3,
            "Baltic Avenue",
            SpaceType.PROPERTY,
            PropertyColor.BROWN,
            60,
            (4, 20, 60, 180, 320, 450),
            50,
            30,
        ),
        TaxSpace(4, "Income Tax", SpaceType.INCOME_TAX, 200),
        RailroadSpace(
            5, "Reading Railroad", SpaceType.RAILROAD, 200, (25, 50, 100, 200), 100
        ),
        PropertySpace(
            6,
            "Oriental Avenue",
            SpaceType.PROPERTY,
            PropertyColor.LIGHT_BLUE,
            100,
            (6, 30, 90, 270, 400, 550),
            50,
            50,
        ),
        Space(7, "Chance", SpaceType.CHANCE),
        PropertySpace(
            8,
            "Vermont Avenue",
            SpaceType.PROPERTY,
            PropertyColor.LIGHT_BLUE,
            100,
            (6, 30, 90, 270, 400, 550),
            50,
            50,
        ),
        PropertySpace(
            9,
            "Connecticut Avenue",
            SpaceType.PROPERTY,
            PropertyColor.LIGHT_BLUE,
            120,
            (8, 40, 100, 300, 450, 600),
            50,
            60,
        ),
        # Left column (bottom to top)
        Space(10, "Jail / Just Visiting", SpaceType.JAIL),
        PropertySpace(
            11,
            "St. Charles Place",
            SpaceType.PROPERTY,
            PropertyColor.MAGENTA,
            140,
            (10, 50, 150, 450, 625, 750),
            100,
            70,
        ),
        UtilitySpace(12, "Electric Company", SpaceType.UTILITY, 150, 75),
        PropertySpace(
            13,
            "States Avenue",
            SpaceType.PROPERTY,
            PropertyColor.MAGENTA,
            140,
            (10, 50, 150, 450, 625, 750),
            100,
            70,
        ),
        PropertySpace(
            14,
            "Virginia Avenue",
            SpaceType.PROPERTY,
            PropertyColor.MAGENTA,
            160,
            (12, 60, 180, 500, 700, 900),
            100,
            80,
        ),
        RailroadSpace(
            15,
            "Pennsylvania Railroad",
            SpaceType.RAILROAD,
            200,
            (25, 50, 100, 200),
            100,
        ),
        PropertySpace(
            16,
            "St. James Place",
            SpaceType.PROPERTY,
            PropertyColor.ORANGE,
            180,
            (14, 70, 200, 550, 750, 950),
            100,
            90,
        ),
        Space(17, "Community Chest", SpaceType.COMMUNITY_CHEST),
        PropertySpace(
            18,
            "Tennessee Avenue",
            SpaceType.PROPERTY,
            PropertyColor.ORANGE,
            180,
            (14, 70, 200, 550, 750, 950),
            100,
            90,
        ),
        PropertySpace(
            19,
            "New York Avenue",
            SpaceType.PROPERTY,
            PropertyColor.ORANGE,
            200,
            (16, 80, 220, 600, 800, 1000),
            100,
            100,
        ),
        # Top row (left to right)
        Space(20, "Free Parking", SpaceType.FREE_PARKING),
        PropertySpace(
            21,
            "Kentucky Avenue",
            SpaceType.PROPERTY,
            PropertyColor.RED,
            220,
            (18, 90, 250, 700, 875, 1050),
            150,
            110,
        ),
        Space(22, "Chance", SpaceType.CHANCE),
        PropertySpace(
            23,
            "Indiana Avenue",
            SpaceType.PROPERTY,
            PropertyColor.RED,
            220,
            (18, 90, 250, 700, 875, 1050),
            150,
            110,
        ),
        PropertySpace(
            24,
            "Illinois Avenue",
            SpaceType.PROPERTY,
            PropertyColor.RED,
            240,
            (20, 100, 300, 750, 925, 1100),
            150,
            120,
        ),
        RailroadSpace(
            25, "B&O Railroad", SpaceType.RAILROAD, 200, (25, 50, 100, 200), 100
        ),
        PropertySpace(
            26,
            "Atlantic Avenue",
            SpaceType.PROPERTY,
            PropertyColor.YELLOW,
            260,
            (22, 110, 330, 800, 975, 1150),
            150,
            130,
        ),
        PropertySpace(
            27,
            "Ventnor Avenue",
            SpaceType.PROPERTY,
            PropertyColor.YELLOW,
            260,
            (22, 110, 330, 800, 975, 1150),
            150,
            130,
        ),
        UtilitySpace(28, "Water Works", SpaceType.UTILITY, 150, 75),
        PropertySpace(
            29,
            "Marvin Gardens",
            SpaceType.PROPERTY,
            PropertyColor.YELLOW,
            280,
            (24, 120, 360, 850, 1025, 1200),
            150,
            140,
        ),
        # Right column (top to bottom)
        Space(30, "Go To Jail", SpaceType.GO_TO_JAIL),
        PropertySpace(
            31,
            "Pacific Avenue",
            SpaceType.PROPERTY,
            PropertyColor.GREEN,
            300,
            (26, 130, 390, 900, 1100, 1275),
            200,
            150,
        ),
        PropertySpace(
            32,
            "North Carolina Avenue",
            SpaceType.PROPERTY,
            PropertyColor.GREEN,
            300,
            (26, 130, 390, 900, 1100, 1275),
            200,
            150,
        ),
        Space(33, "Community Chest", SpaceType.COMMUNITY_CHEST),
        PropertySpace(
            34,
            "Pennsylvania Avenue",
            SpaceType.PROPERTY,
            PropertyColor.GREEN,
            320,
            (28, 150, 450, 1000, 1200, 1400),
            200,
            160,
        ),
        RailroadSpace(
            35, "Short Line", SpaceType.RAILROAD, 200, (25, 50, 100, 200), 100
        ),
        Space(36, "Chance", SpaceType.CHANCE),
        PropertySpace(
            37,
            "Park Place",
            SpaceType.PROPERTY,
            PropertyColor.DARK_BLUE,
            350,
            (35, 175, 500, 1100, 1300, 1500),
            200,
            175,
        ),
        TaxSpace(38, "Luxury Tax", SpaceType.LUXURY_TAX, 100),
        PropertySpace(
            39,
            "Boardwalk",
            SpaceType.PROPERTY,
            PropertyColor.DARK_BLUE,
            400,
            (50, 200, 600, 1400, 1700, 2000),
            200,
            200,
        ),
    )

    # Pre-computed indices for quick lookups
    _property_positions: ClassVar[tuple[int, ...]] = tuple(
        i for i, s in enumerate(SPACES) if isinstance(s, PropertySpace)
    )
    _railroad_positions: ClassVar[tuple[int, ...]] = (5, 15, 25, 35)
    _utility_positions: ClassVar[tuple[int, ...]] = (12, 28)
    _chance_positions: ClassVar[tuple[int, ...]] = (7, 22, 36)
    _chest_positions: ClassVar[tuple[int, ...]] = (2, 17, 33)

    @classmethod
    def get_space(cls, position: int) -> Space:
        """Get the space at a given position (wraps around)."""
        return cls.SPACES[position % 40]

    @classmethod
    def get_property_group(cls, color: PropertyColor) -> tuple[int, ...]:
        """Get all positions for a property color group."""
        return PROPERTY_GROUPS.get(color, ())

    @classmethod
    def get_group_size(cls, color: PropertyColor) -> int:
        """Get the number of properties in a color group."""
        return len(PROPERTY_GROUPS.get(color, ()))

    @classmethod
    def is_property(cls, position: int) -> bool:
        """Check if position is a regular property."""
        return position in cls._property_positions

    @classmethod
    def is_railroad(cls, position: int) -> bool:
        """Check if position is a railroad."""
        return position in cls._railroad_positions

    @classmethod
    def is_utility(cls, position: int) -> bool:
        """Check if position is a utility."""
        return position in cls._utility_positions

    @classmethod
    def is_buyable(cls, position: int) -> bool:
        """Check if the space at this position can be purchased."""
        space = cls.get_space(position)
        return isinstance(space, (PropertySpace, RailroadSpace, UtilitySpace))

    @classmethod
    def get_buyable_positions(cls) -> tuple[int, ...]:
        """Get all positions that can be purchased."""
        return tuple(
            i
            for i, s in enumerate(cls.SPACES)
            if isinstance(s, (PropertySpace, RailroadSpace, UtilitySpace))
        )

    @classmethod
    def get_nearest_railroad(cls, position: int) -> int:
        """Get the nearest railroad position (moving forward)."""
        for rr in cls._railroad_positions:
            if rr > position:
                return rr
        return cls._railroad_positions[0]  # Wrap around to Reading Railroad

    @classmethod
    def get_nearest_utility(cls, position: int) -> int:
        """Get the nearest utility position (moving forward)."""
        for util in cls._utility_positions:
            if util > position:
                return util
        return cls._utility_positions[0]  # Wrap around to Electric Company

    @classmethod
    def calculate_distance(cls, from_pos: int, to_pos: int) -> int:
        """Calculate spaces between two positions (moving forward)."""
        if to_pos >= from_pos:
            return to_pos - from_pos
        return 40 - from_pos + to_pos

    @classmethod
    def get_all_spaces(cls) -> tuple[Space, ...]:
        """Get all spaces on the board."""
        return cls.SPACES

    @classmethod
    def to_dict(cls) -> dict[str, object]:
        """Serialize the board to a JSON-compatible dict."""
        return {
            "spaces": [space.to_dict() for space in cls.SPACES],
            "property_groups": {
                color.name: list(positions)
                for color, positions in PROPERTY_GROUPS.items()
            },
        }

File: frontend/src/utils/board.ts
import type { BoardSpace, PropertyInfo, PropertyColor } from '@/types';

/**
 * Board layout positions for CSS Grid rendering.
 * Each array contains the positions for that side of the board.
 */
export const BOARD_LAYOUT = {
  // Bottom row: GO (0) to Jail (10) - right to left
  bottom: [10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0],
  // Left column: Connecticut (9) up to St. Charles (11) - bottom to top
  left: [11, 12, 13, 14, 15, 16, 17, 18, 19],
  // Top row: Free Parking (20) to Go To Jail (30) - left to right
  top: [20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30],
  // Right column: Pacific (31) down to Park Place (39) - top to bottom
  right: [31, 32, 33, 34, 35, 36, 37, 38, 39],
};

/**
 * All 40 board spaces with their information.
 */
export const BOARD_SPACES: BoardSpace[] = [
  // Bottom row (0-10)
  { position: 0, name: 'GO', type: 'corner' },
  { position: 1, name: 'Mediterranean Avenue', type: 'property', color: 'brown' },
  { position: 2, name: 'Community Chest', type: 'community_chest' },
  { position: 3, name: 'Baltic Avenue', type: 'property', color: 'brown' },
  { position: 4, name: 'Income Tax', type: 'tax' },
  { position: 5, name: 'Reading Railroad', type: 'railroad' },
  { position: 6, name: 'Oriental Avenue', type: 'property', color: 'lightblue' },
  { position: 7, name: 'Chance', type: 'chance' },
  { position: 8, name: 'Vermont Avenue', type: 'property', color: 'lightblue' },
  { position: 9, name: 'Connecticut Avenue', type: 'property', color: 'lightblue' },
  { position: 10, name: 'Jail / Just Visiting', type: 'corner' },

  // Left column (11-19)
  { position: 11, name: 'St. Charles Place', type: 'property', color: 'magenta' },
  { position: 12, name: 'Electric Company', type: 'utility' },
  { position: 13, name: 'States Avenue', type: 'property', color: 'magenta' },
  { position: 14, name: 'Virginia Avenue', type: 'property', color: 'magenta' },
  { position: 15, name: 'Pennsylvania Railroad', type: 'railroad' },
  { position: 16, name: 'St. James Place', type: 'property', color: 'orange' },
  { position: 17, name: 'Community Chest', type: 'community_chest' },
  { position: 18, name: 'Tennessee Avenue', type: 'property', color: 'orange' },
  { position: 19, name: 'New York Avenue', type: 'property', color: 'orange' },

  // Top row (20-30)
  { position: 20, name: 'Free Parking', type: 'corner' },
  { position: 21, name: 'Kentucky Avenue', type: 'property', color: 'red' },
  { position: 22, name: 'Chance', type: 'chance' },
  { position: 23, name: 'Indiana Avenue', type: 'property', color: 'red' },
  { position: 24, name: 'Illinois Avenue', type: 'property', color: 'red' },
  { position: 25, name: 'B&O Railroad', type: 'railroad' },
  { position: 26, name: 'Atlantic Avenue', type: 'property', color: 'yellow' },
  { position: 27, name: 'Ventnor Avenue', type: 'property', color: 'yellow' },
  { position: 28, name: 'Water Works', type: 'utility' },
  { position: 29, name: 'Marvin Gardens', type: 'property', color: 'yellow' },
  { position: 30, name: 'Go To Jail', type: 'corner' },

  // Right column (31-39)
  { position: 31, name: 'Pacific Avenue', type: 'property', color: 'green' },
  { position: 32, name: 'North Carolina Avenue', type: 'property', color: 'green' },
  { position: 33, name: 'Community Chest', type: 'community_chest' },
  { position: 34, name: 'Pennsylvania Avenue', type: 'property', color: 'green' },
  { position: 35, name: 'Short Line', type: 'railroad' },
  { position: 36, name: 'Chance', type: 'chance' },
  { position: 37, name: 'Park Place', type: 'property', color: 'blue' },
  { position: 38, name: 'Luxury Tax', type: 'tax' },
  { position: 39, name: 'Boardwalk', type: 'property', color: 'blue' },
];

/**
 * Property information including prices and rents.
 */
export const PROPERTY_INFO: Record<number, PropertyInfo> = {
  1: {
    position: 1,
    name: 'Mediterranean Avenue',
    type: 'property',
    color: 'brown',
    price: 60,
    rent: [2, 10, 30, 90, 160, 250],
    buildCost: 50,
    mortgageValue: 30,
  },
  3: {
    position: 3,
    name: 'Baltic Avenue',
    type: 'property',
    color: 'brown',
    price: 60,
    rent: [4, 20, 60, 180, 320, 450],
    buildCost: 50,
    mortgageValue: 30,
  },
  5: {
    position: 5,
    name: 'Reading Railroad',
    type: 'railroad',
    price: 200,
    rent: [25, 50, 100, 200],
    buildCost: 0,
    mortgageValue: 100,
  },
  6: {
    position: 6,
    name: 'Oriental Avenue',
    type: 'property',
    color: 'lightblue',
    price: 100,
    rent: [6, 30, 90, 270, 400, 550],
    buildCost: 50,
    mortgageValue: 50,
  },
  8: {
    position: 8,
    name: 'Vermont Avenue',
    type: 'property',
    color: 'lightblue',
    price: 100,
    rent: [6, 30, 90, 270, 400, 550],
    buildCost: 50,
    mortgageValue: 50,
  },
  9: {
    position: 9,
    name: 'Connecticut Avenue',
    type: 'property',
    color: 'lightblue',
    price: 120,
    rent: [8, 40, 100, 300, 450, 600],
    buildCost: 50,
    mortgageValue: 60,
  },
  11: {
    position: 11,
    name: 'St. Charles Place',
    type: 'property',
    color: 'magenta',
    price: 140,
    rent: [10, 50, 150, 450, 625, 750],
    buildCost: 100,
    mortgageValue: 70,
  },
  12: {
    position: 12,
    name: 'Electric Company',
    type: 'utility',
    price: 150,
    rent: [4, 10], // Multiplied by dice roll
    buildCost: 0,
    mortgageValue: 75,
  },
  13: {
    position: 13,
    name: 'States Avenue',
    type: 'property',
    color: 'magenta',
    price: 140,
    rent: [10, 50, 150, 450, 625, 750],
    buildCost: 100,
    mortgageValue: 70,
  },
  14: {
    position: 14,
    name: 'Virginia Avenue',
    type: 'property',
    color: 'magenta',
    price: 160,
    rent: [12, 60, 180, 500, 700, 900],
    buildCost: 100,
    mortgageValue: 80,
  },
  15: {
    position: 15,
    name: 'Pennsylvania Railroad',
    type: 'railroad',
    price: 200,
    rent: [25, 50, 100, 200],
    buildCost: 0,
    mortgageValue: 100,
  },
  16: {
    position: 16,
    name: 'St. James Place',
    type: 'property',
    color: 'orange',
    price: 180,
    rent: [14, 70, 200, 550, 750, 950],
    buildCost: 100,
    mortgageValue: 90,
  },
  18: {
    position: 18,
    name: 'Tennessee Avenue',
    type: 'property',
    color: 'orange',
    price: 180,
    rent: [14, 70, 200, 550, 750, 950],
    buildCost: 100,
    mortgageValue: 90,
  },
  19: {
    position: 19,
    name: 'New York Avenue',
    type: 'property',
    color: 'orange',
    price: 200,
    rent: [16, 80, 220, 600, 800, 1000],
    buildCost: 100,
    mortgageValue: 100,
  },
  21: {
    position: 21,
    name: 'Kentucky Avenue',
    type: 'property',
    color: 'red',
    price: 220,
    rent: [18, 90, 250, 700, 875, 1050],
    buildCost: 150,
    mortgageValue: 110,
  },
  23: {
    position: 23,
    name: 'Indiana Avenue',
    type: 'property',
    color: 'red',
    price: 220,
    rent: [18, 90, 250, 700, 875, 1050],
    buildCost: 150,
    mortgageValue: 110,
  },
  24: {
    position: 24,
    name: 'Illinois Avenue',
    type: 'property',
    color: 'red',
    price: 240,
    rent: [20, 100, 300, 750, 925, 1100],
    buildCost: 150,
    mortgageValue: 120,
  },
  25: {
    position: 25,
    name: 'B&O Railroad',
    type: 'railroad',
    price: 200,
    rent: [25, 50, 100, 200],
    buildCost: 0,
    mortgageValue: 100,
  },
  26: {
    position: 26,
    name: 'Atlantic Avenue',
    type: 'property',
    color: 'yellow',
    price: 260,
    rent: [22, 110, 330, 800, 975, 1150],
    buildCost: 150,
    mortgageValue: 130,
  },
  27: {
    position: 27,
    name: 'Ventnor Avenue',
    type: 'property',
    color: 'yellow',
    price: 260,
    rent: [22, 110, 330, 800, 975, 1150],
    buildCost: 150,
    mortgageValue: 130,
  },
  28: {
    position: 28,
    name: 'Water Works',
    type: 'utility',
    price: 150,
    rent: [4, 10], // Multiplied by dice roll
    buildCost: 0,
    mortgageValue: 75,
  },
  29: {
    position: 29,
    name: 'Marvin Gardens',
    type: 'property',
    color: 'yellow',
    price: 280,
    rent: [24, 120, 360, 850, 1025, 1200],
    buildCost: 150,
    mortgageValue: 140,
  },
  31: {
    position: 31,
    name: 'Pacific Avenue',
    type: 'property',
    color: 'green',
    price: 300,
    rent: [26, 130, 390, 900, 1100, 1275],
    buildCost: 200,
    mortgageValue: 150,
  },
  32: {
    position: 32,
    name: 'North Carolina Avenue',
    type: 'property',
    color: 'green',
    price: 300,
    rent: [26, 130, 390, 900, 1100, 1275],
    buildCost: 200,
    mortgageValue: 150,
  },
  34: {
    position: 34,
    name: 'Pennsylvania Avenue',
    type: 'property',
    color: 'green',
    price: 320,
    rent: [28, 150, 450, 1000, 1200, 1400],
    buildCost: 200,
    mortgageValue: 160,
  },
  35: {
    position: 35,
    name: 'Short Line',
    type: 'railroad',
    price: 200,
    rent: [25, 50, 100, 200],
    buildCost: 0,
    mortgageValue: 100,
  },
  37: {
    position: 37,
    name: 'Park Place',
    type: 'property',
    color: 'blue',
    price: 350,
    rent: [35, 175, 500, 1100, 1300, 1500],
    buildCost: 200,
    mortgageValue: 175,
  },
  39: {
    position: 39,
    name: 'Boardwalk',
    type: 'property',
    color: 'blue',
    price: 400,
    rent: [50, 200, 600, 1400, 1700, 2000],
    buildCost: 200,
    mortgageValue: 200,
  },
};

/**
 * Get board space by position.
 */
export function getBoardSpace(position: number): BoardSpace | undefined {
  return BOARD_SPACES.find((space) => space.position === position);
}

/**
 * Get property info by position.
 */
export function getPropertyInfo(position: number): PropertyInfo | undefined {
  return PROPERTY_INFO[position];
}

/**
 * Get positions of all properties in a color group.
 */
export function getColorGroupPositions(color: PropertyColor): number[] {
  return BOARD_SPACES.filter((space) => space.color === color).map((space) => space.position);
}

/**
 * Check if a position is a corner space.
 */
export function isCornerSpace(position: number): boolean {
  return [0, 10, 20, 30].includes(position);
}

/**
 * Get side of board for a position.
 */
export function getBoardSide(position: number): 'bottom' | 'left' | 'top' | 'right' {
  if (position <= 10) return 'bottom';
  if (position <= 19) return 'left';
  if (position <= 30) return 'top';
  return 'right';
}

/**
 * Get CSS color class for property color.
 */
export function getColorClass(color?: PropertyColor): string {
  if (!color) return '';

  const colorMap: Record<PropertyColor, string> = {
    brown: 'bg-property-brown',
    lightblue: 'bg-property-lightblue',
    magenta: 'bg-property-magenta',
    orange: 'bg-property-orange',
    red: 'bg-property-red',
    yellow: 'bg-property-yellow',
    green: 'bg-property-green',
    blue: 'bg-property-blue',
    railroad: 'bg-property-railroad',
    utility: 'bg-property-utility',
  };

  return colorMap[color] || '';
}

/**
 * Get hex color for property color.
 */
export function getColorHex(color?: PropertyColor): string {
  if (!color) return '#808080';

  const colorMap: Record<PropertyColor, string> = {
    brown: '#8B4513',
    lightblue: '#87CEEB',
    magenta: '#FF00FF',
    orange: '#FFA500',
    red: '#FF0000',
    yellow: '#FFFF00',
    green: '#008000',
    blue: '#0000FF',
    railroad: '#000000',
    utility: '#808080',
  };

  return colorMap[color];
}
